modules/base/rendering/renderablenodedirectionhint.cpp

```diff
@@ -0,0 +1,544 @@
+        "If 'UseRelativeOffset' is true, the value should be given as a factor to "
+        "multiply with the boudning sphere of the node. Otherwise, the value is "
+        "specified in meters",
+        openspace::properties::Property::Visibility::AdvancedUser
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo RelativeOffsetInfo = {
+        "UseRelativeOffset",
+        "Use Relative Offset Distance",
+        "Decide whether to use relative distances (in units of start node bounding "
+        "sphere) for the offset distance. If false, meters is used",
+        openspace::properties::Property::Visibility::AdvancedUser
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LengthInfo = {
+        "Length",
+        "Length",
+        "The length of the arrow, given either in meters or as a factor to be "
+        "multiplied with the bounding sphere of the start node (if "
+        "'UseRelativeLength' is true)",
+        openspace::properties::Property::Visibility::AdvancedUser
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo RelativeLengthInfo = {
+        "UseRelativeLength",
+        "Use Relative Length",
+        "Decide whether to use relative size (in units of start node bounding "
+        "sphere) for the length of the arrow. If false, meters is used",
+        openspace::properties::Property::Visibility::AdvancedUser
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo WidthInfo = {
+        "Width",
+        "Width",
+        "This value specifies the width of the arrow shape",
+        openspace::properties::Property::Visibility::AdvancedUser
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo AmbientIntensityInfo = {
+        "AmbientIntensity",
+        "Ambient Intensity",
+        "A multiplier for ambient lighting",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo DiffuseIntensityInfo = {
+        "DiffuseIntensity",
+        "Diffuse Intensity",
+        "A multiplier for diffuse lighting",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo SpecularIntensityInfo = {
+        "SpecularIntensity",
+        "Specular Intensity",
+        "A multiplier for specular lighting",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo ShadingEnabledInfo = {
+        "PerformShading",
+        "Perform Shading",
+        "This value determines whether the arrow model should be shaded by using the "
+        "position of the Sun",
+        openspace::properties::Property::Visibility::User
+    };
+
+    struct [[codegen::Dictionary(RenderableNodeDirectionHint)]] Parameters {
+        // [[codegen::verbatim(StartNodeInfo.description)]]
+        std::optional<std::string> startNode [[codegen::notempty]];
```

Comment:
Seems to me like the `startNode` and `endNode` would not be optional?

Reply:
No, you're probably right.  Then there is no point in saying that it defaults to Root in the descriptions either